lib/src/lib.rs

```diff
@@ -389,15 +387,16 @@ impl<'a> ChunkAccessor for DefaultChunkAccessor<'a> {
             data
         };
 
-        let mut sha256 = sha2::Sha256::new();
+        let mut sha256 = Sha256::default();
         sha256.input(&data);
-        let mut sha256_digest = vec![0u8; DIGEST_SIZE];
-        sha256.result(&mut sha256_digest);
-        if sha256_digest != digest {
+        let mut vec_result = vec![0u8; DIGEST_SIZE];
+        vec_result.copy_from_slice(&sha256.result());
```

Comment:
This digest needs to be sent to another thread, so I guess `Vec` is indeed required. Thanks!